### ISSUE TITLE
NullPointerException with create tenant endpoint

### ISSUE THREAD

**Issue by JEduardogmr** [FusionAuth/fusionauth-issues#93]
### Description 
When i create a tenant with a malformed request body i receive a NullPointerException and a 500 from the API.

### Steps to reproduce
Assuming the Authorization header is added with the api key:

1. Perform a POST request to /api/tenant with request body
```js
{
  name: "test"
}
```

### Expected behavior
A 500 response is retrieved by the API, instead of a 400.

Response body:
```js
{
    "generalErrors": [
        {
            "code": "[Exception]",
            "message": "FusionAuth encountered an unexpected error. Please contact support for assistance."
        }
    ]
}
````

Here is the stack trace of the exception:

```js
io.fusionauth.app.primeframework.error.ExceptionExceptionHandler - An unhandled exception was thrown java.lang.NullPointerException: null
at io.fusionauth.app.action.api.TenantAction.validate(TenantAction.java:87) ~[fusionauth-app-1.4.0.jar:1.4.0]
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_171]
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_171]
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_171]
at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_171]
at org.primeframework.mvc.util.ReflectionUtils.invoke(ReflectionUtils.java:436) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.validation.DefaultValidationProcessor.validate(DefaultValidationProcessor.java:78) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.validation.DefaultValidationWorkflow.perform(DefaultValidationWorkflow.java:46) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.security.DefaultSecurityWorkflow.perform(DefaultSecurityWorkflow.java:81) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.parameter.DefaultPostParameterWorkflow.perform(DefaultPostParameterWorkflow.java:50) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.content.DefaultContentWorkflow.perform(DefaultContentWorkflow.java:52) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.parameter.DefaultParameterWorkflow.perform(DefaultParameterWorkflow.java:57) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.parameter.DefaultURIParameterWorkflow.perform(DefaultURIParameterWorkflow.java:102) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.scope.DefaultScopeRetrievalWorkflow.perform(DefaultScopeRetrievalWorkflow.java:58) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.message.DefaultMessageWorkflow.perform(DefaultMessageWorkflow.java:45) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.action.DefaultActionMappingWorkflow.perform(DefaultActionMappingWorkflow.java:126) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.StaticResourceWorkflow.perform(StaticResourceWorkflow.java:97) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.parameter.RequestBodyWorkflow.perform(RequestBodyWorkflow.java:89) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.security.DefaultSavedRequestWorkflow.perform(DefaultSavedRequestWorkflow.java:57) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.SubWorkflowChain.continueWorkflow(SubWorkflowChain.java:43) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.DefaultMVCWorkflow.perform(DefaultMVCWorkflow.java:91) ~[prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.workflow.DefaultWorkflowChain.continueWorkflow(DefaultWorkflowChain.java:44) [prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.servlet.FilterWorkflowChain.continueWorkflow(FilterWorkflowChain.java:50) [prime-mvc-1.12.0.jar:1.12.0]
at org.primeframework.mvc.servlet.PrimeFilter.doFilter(PrimeFilter.java:84) [prime-mvc-1.12.0.jar:1.12.0]
at com.inversoft.maintenance.servlet.MaintenanceModePrimeFilter.doFilter(MaintenanceModePrimeFilter.java:59) [inversoft-maintenance-mode-0.12.8.jar:0.12.8]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.31]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.31]
at com.inversoft.servlet.CORSFilter.handleNonCORS(CORSFilter.java:748) [inversoft-servlet-0.1.1.jar:0.1.1]
at com.inversoft.servlet.CORSFilter.doFilter(CORSFilter.java:646) [inversoft-servlet-0.1.1.jar:0.1.1]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.31]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.31]
at com.inversoft.servlet.UTF8Filter.doFilter(UTF8Filter.java:27) [inversoft-servlet-0.1.1.jar:0.1.1]
at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.31]
at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.31]
at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:198) [catalina.jar:8.5.31]
at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [catalina.jar:8.5.31]
at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:496) [catalina.jar:8.5.31]
at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:140) [catalina.jar:8.5.31]
at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.31]
at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) [catalina.jar:8.5.31]
at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:342) [catalina.jar:8.5.31]
at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:803) [tomcat-coyote.jar:8.5.31]
at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:66) [tomcat-coyote.jar:8.5.31]
at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:790) [tomcat-coyote.jar:8.5.31]
at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1468) [tomcat-coyote.jar:8.5.31]
at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.31]
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [na:1.8.0_171]
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [na:1.8.0_171]
at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.31]
at java.lang.Thread.run(Thread.java:748) [na:1.8.0_171]
````

### Platform
 - Device: Desktop
 - OS: macOS
 - Postman

### Additional context
FusionAuth is running on docker container.

**Comment by robotdan** [comment]
Thanks for reporting this issue @JEduardogmr. This is a known issue, it has been resolved and will be fixed in the upcoming version `1.5.0`. 

When you send that request body, are you also sending the `Content-Type` header with value `application/json`? 

If you send the `Content-Type` header, I would expect you see a different error, for example: 

```
{
    "generalErrors": [
        {
            "code": "[invalidJSON]",
            "message": "Invalid JSON in the request body. The property was [name]. The error was [Unrecognized property]. The detailed exception was [Unrecognized field \"name\" (class io.fusionauth.domain.api.TenantRequest), not marked as ignorable (one known property: \"tenant\"])\n at [Source: (org.apache.catalina.connector.CoyoteInputStream); line: 3, column: 11] (through reference chain: io.fusionauth.domain.api.TenantRequest[\"name\"])]."
        }
    ]
}
```

**Comment by JEduardogmr** [comment]
Hi @robotdan. That's good news!

Regarding your question, yes i'm sending the `Content-Type` header as  `application/json` and i'm receiving the response i reported.

**Comment by robotdan** [comment]
This issue has been resolved and available in version 1.5.0. 
https://fusionauth.io/docs/v1/tech/release-notes/

Thanks for taking the time to report this issue @JEduardogmr!
